privacy_consent/models/privacy_consent.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Tecnativa - Jairo Llopis
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import hashlib
+import hmac
+
+from odoo import api, fields, models
+
+
+class PrivacyConsent(models.Model):
+    _name = 'privacy.consent'
+    _description = "Consent of data processing"
+    _inherit = "mail.thread"
+    _rec_name = "partner_id"
+    _sql_constraints = [
+        ("unique_partner_activity", "UNIQUE(partner_id, activity_id)",
+         "Duplicated partner in this data processing activity"),
+    ]
+
+    active = fields.Boolean(
+        default=True,
+        index=True,
+    )
+    accepted = fields.Boolean(
+        track_visibility="onchange",
+        help="Indicates current acceptance status, which can come from "
+             "subject's last answer, or from the default specified in the "
+             "related data processing activity.",
+    )
+    last_metadata = fields.Text(
+        readonly=True,
+        track_visibility="onchange",
+        help="Metadata from the last acceptance or rejection by the subject",
+    )
+    partner_id = fields.Many2one(
```

Comment:
Wouldn't this be more extensible to other models (mass_mailing.contact, event.registration) with a `Reference` field?

Reply:
It was my initial approach, but it turns out that Odoo always creates a partner the 1st time it sends an email to it, or even to be able to suggest it as a follower.

Since, at the end of the day, a partner was created, I found it easier to just use partners. If you need other models, just create the partners automatically through other mechanism, and you're done.